editor/src/components/shared/scoped-variables.ts

```diff
@@ -131,9 +133,59 @@ function getMatchingElementForVariable(variable: Variable) {
       return jsxElementWithoutUID('span', jsxAttributesFromMap({}), [
         jsExpressionOtherJavaScriptSimple(`JSON.stringify(${variable.name})`, [variable.name]),
       ])
+    case 'array':
+      return jsxFragmentWithoutUID(
+        [jsxTextBlock(`{${variable.name}.map((item) => <span>{JSON.stringify(item)}</span>)}`)],
```

Comment:
Nice. I guess the next step is to have this recursively call `getMatchingElementForVariable` then?

Reply:
No, since we can't use the actual element here (since it's a text block). I agree it's better to do it recursively, it just means that all of the `switch-case` branches will have to be text (for example - the conditional can't be an element but has to be `${variable.name} ? null : null` as a template string)